src/api/java/com/minecolonies/api/colony/requestsystem/requester/IRequesterFactory.java

```diff
@@ -0,0 +1,11 @@
+package com.minecolonies.api.colony.requestsystem.requester;
+
+import com.minecolonies.api.colony.requestsystem.factory.IFactory;
+
+/**
+ * Interface describing objects that can construct IRequester objects.
+ */
+public interface IRequesterFactory<Input, Output extends IRequester> extends IFactory<Input, Output>
```

Comment:
This class is not supposed to be in this PR yet.
Just the Factories part and the IToken part (and there respective implementations)